### PR TITLE
added requirements

### PR DESCRIPTION
**Pull request by boroda4436**

### CODE REVIEW

File: src/main/java/mate/academy/dao/impl/RoleDaoImpl.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+import mate.academy.exception.DataProcessingException;
+import mate.academy.model.Role;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class RoleDaoImpl extends AbstractDao<Role, Long> implements RoleDao {
+
```
Comment (Resci) [inline src/main/java/mate/academy/dao/impl/RoleDaoImpl.java:14]:
```suggestion
```

File: src/main/java/mate/academy/security/AuthenticationServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        user.setRoles(Set.of(roleService.getRoleByName("USER")));
+        user = userService.save(user);
+        return user;
+    }
+
+    @Override
+    public User login(String login, String password) throws AuthenticationException {
+        Optional<User> user = userService.findByEmail(login);
+        String encodedPassword = passwordEncoder.encode(password);
+        if (user.isEmpty() || user.get().getPassword().equals(encodedPassword)) {
```
Comment (Resci) [inline src/main/java/mate/academy/security/AuthenticationServiceImpl.java:40]:
```suggestion
        if (user.isEmpty() || !passwordEncoder.matches(password, user.get().getPassword())) {
```

Comment (Resci) [inline src/main/java/mate/academy/security/AuthenticationServiceImpl.java:40]:
```suggestion
        if (user.isEmpty() || !passwordEncoder.matches(password, user.get().getPassword())) {
```

File: pom.xml [hunk 3]
```diff
@@ -5,47 +5,114 @@
...
-                <artifactId>maven-checkstyle-plugin</artifactId>
-                <version>3.1.1</version>
-                <executions>
-                    <execution>
-                        <phase>compile</phase>
-                        <goals>
-                            <goal>check</goal>
-                        </goals>
-                    </execution>
-                </executions>
```
[the remark this answers is not in the document]
Comment (boroda4436) [reply pom.xml:42]: Good point! Thanks!

File: src/main/java/mate/academy/security/AuthenticationServiceImpl.java (hunk 2 above)
Comment (boroda4436) [reply src/main/java/mate/academy/security/AuthenticationServiceImpl.java]: fixed

**Review by Resci — APPROVED** [review]

**Pull request merged**